src/sagemaker_server.cc

```diff
@@ -670,6 +732,35 @@ SagemakerAPIServer::SageMakerMMEUnloadModel(
 
   HandleRepositoryControl(req, "", model_name, "unload");
 
+  /*Note: Model status check is repo-specific and therefore must be run before
+   * unregistering the repo, else the model information is lost*/
+  bool is_model_unavailable = false;
+  uint32_t unload_time_in_secs = 0;
+
+  LOG_VERBOSE(1) << "Using Model Repository Index during UNLOAD to check for "
+                    "status of model: "
+                 << model_name;
+
+  /* Wait for the model to be completely unloaded. SageMaker waits a maximum of
+  360 seconds for the UNLOAD request to timeout. Setting a limit of 350 seconds
+  for Triton unload.*/
+  while (!is_model_unavailable && unload_time_in_secs < UNLOAD_TIMEOUT_SECS_) {
+    is_model_unavailable = SageMakerMMEUnloadModelCheckStatus(model_name);
+    sleep(1);
+    unload_time_in_secs += 1;
```

Comment:
I believe this does not account for time taken for this `SageMakerMMEUnloadModelCheckStatus(model_name)` function call. Is it too small to be neglected?

Reply:
That's correct, in SM-Triton, there's one model repo per model, so the model repository index is expected to contain only one model (except in case of ensembles, where it maybe approx ~5-8), so the call is expected to return within few milliseconds.

Reply:
IMHO, we can measure time for that as well and report a more accurate number. It may not matter much in normal cases but if something weird happens in that function, we will be logging incorrect unload time.

Reply:
Related issue: `sleep` is not exact, either. 

Rather than doing `unload_time_in_secs += 1`, we should be measuring elapsed time since start of loop after each sleep. This would solve both problems.

Reply:
Thank you for comments, I've addressed this by using elapsed time calculation.

Reply:
> so the model repository index is expected to contain only one model.

Note that index API poll all registered model repos, so the performance is actually proportioned to all "visible" models. Although that is probably still neglectable with respect to the unload time.